internal/extramanifest/policy.go

```diff
@@ -0,0 +1,125 @@
+package extramanifest
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	policyv1 "open-cluster-management.io/config-policy-controller/api/v1"
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//+kubebuilder:rbac:groups=policy.open-cluster-management.io,resources=policies,verbs=get;list
+
+const ztpDeployWaveAnnotation = "ran.openshift.io/ztp-deploy-wave"
+
+// GetPolicies gets the policies matching the labels from the namespace
+func (h *EMHandler) GetPolicies(ctx context.Context, labels map[string]string) ([]*policiesv1.Policy, error) {
+	listOpts := []client.ListOption{
+		client.MatchingLabels(labels),
+	}
+
+	policies := &policiesv1.PolicyList{}
+	if err := h.Client.List(ctx, policies, listOpts...); err != nil {
+		return nil, err
+	}
+
+	var policyWaveMap = make(map[*policiesv1.Policy]int)
+
+	for _, policy := range policies.Items {
+		// Ignore policies with remediationAction enforce
+		if strings.EqualFold(string(policy.Spec.RemediationAction), "enforce") {
+			h.Log.Info("Ignoring policy " + policy.Name + " with remediationAction enforce")
+			continue
+		}
+
+		deployWave, found := policy.GetAnnotations()[ztpDeployWaveAnnotation]
```

Comment:
Do we simply ignore all policies without this wave annotation? Is this the intention? 

If all policies should have wave, should this maybe be an error instead?

Reply:
The annotation helps with the ordering as well as whether the policy should be included in the auto created CGU post cluster installation. Our reference du profile policies all have it and we ask customers do the same to their policies if they are meant to be included.

Reply:
So all policies have it, then my question remains:

> should this maybe be an error instead?

Reply:
> So all policies have it, then my question remains:
> 
> > should this maybe be an error instead?

All policies that should be auto applied as part of new install or upgrade should have it. It's possible that other policies are placed to some cluster temporarily.

Reply:
> I don't think we can guarantee that customer provided policies would have the wave annotation.

That's our current requirement if customer wants to include their policy as part of the ZTP CGU.

Reply:
if we require all inform policies to have a wave then a policy without a wave is an error